text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,659 @@
+  +------------+                             +------------+
+```
+
+#### Single message producer, single message push-based consumer with manual settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+  +------------+                             +------------------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1       |
+  +------------+                             +-----+-----------++
+                                                   | Settle m1 |
+                                                   +-----------+
+```
+
+#### Single message producer, single message push-based consumer with auto-settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+                                           +--------------------------+
+                                           | Ambient                  |
+  +------------+                           +-+------------+-----------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1 |
+  +------------+                             +------------+-----------+
+                                                          | Settle m1 |
+                                                          +-----------+
+```
+
+#### Batch message producer with "Create" spans, single message pull-based consumer
+
+```
+  PRODUCER                                        CONSUMER
+
+  +-------------------------+                     +------------+
+  | Publish                 |         . . . . . . | Receive m1 |
+  +-+-----------+-----------+         .           +------------+
+    | Create m1 | . . . . . . . . . . .
+    +-----------+-----------+                     +------------+
+                | Create m2 | . . . . . . . . . . | Receive m2 |
+                +-----------+                     +------------+
+```
+
+#### Batch message producer, single message push-based consumer
+
+```
+  PRODUCER                                CONSUMER
+
+  +---------+                             +------------+
+  | Publish | . . . . . . . . . . . . . . | Deliver m1 |
+  +---------+                             +------------+
+        .                                 +------------+
+        . . . . . . . . . . . . . . . . . | Deliver m2 |
+                                          +------------+
+```
+
+#### Batch message producer with independent "Create" spans, single message pull-based consumer
+
+```
+  PRODUCER                                      CONSUMER
+
+  +-----------------------------------+
+  | Ambient                           |
+  +-+-----------+---------------------+         +------------+
+    | Create m1 | . . . . . . . . . . . . . . . | Receive m1 |
+    +-----------+-----------+                   +------------+
+                | Create m2 | . . . . . . . .
+                +-----------+---------+     .   +------------+
+                            | Publish |     . . | Receive m2 |
+                            +---------+         +------------+
```

Comment:
Why "Publish" span is now a child of the last "Create" span here?

Reply:
It shows the case where messages arrive at 'Publish' with the context already inside the messages. That's what the `with independent "Create" spans" means. 

The example a bit above "Batch message producer with "Create" spans, single message pull-based consumer" shows the opposite, where the messages arrive at 'Publish' without a context and 'Publish' adds it.

Reply:
This is to highlight the use-case where messages arrive at "Publish" with the create context already populated in them. That's what the title tries to convey: "..  with independent "Create" spans". Maybe it should say something like:

"Batch message producer with "Create" spans populated before publish, single message pull-based consumer"

Reply:
Yeah I agree, it makes much more sense and clear than saying "independent Create spans" imho.

Reply:
Lol, sorry I had commented it before and lost my tab and forgot. Fridays.. 😅

Reply:
Addressed in: [5fa715d](https://github.com/open-telemetry/oteps/pull/192/commits/5fa715dcc6a4e6adafce7fc4032d7f52c24fd31a)